packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,158 @@
+import * as RadixCheckbox from "@radix-ui/react-checkbox";
+import * as RadixLabel from "@radix-ui/react-label";
+import { type ComponentPropsWithRef, forwardRef } from "react";
+
+import type { ExtendProps } from "../utils";
+
+import { Box } from "../box";
+import { Flex } from "../flex";
+import { Text } from "../text";
+import * as styles from "./Checkbox.css";
+
+type CheckboxProps = ExtendProps<
+  ComponentPropsWithRef<typeof Box>,
+  {
+    checked?: "indeterminate" | boolean;
+    defaultChecked?: "indeterminate" | boolean;
+    disabled?: boolean;
+    helperText?: string;
+    label: string;
+    readonly?: boolean;
+  }
+>;
+
+export const Checkbox = forwardRef<HTMLDivElement, CheckboxProps>(
+  (
+    {
+      checked,
+      className,
+      defaultChecked,
+      disabled,
+      helperText,
+      id,
+      label,
+      readonly = false,
+      ...props
+    },
+    ref,
+  ) => {
+    const checkboxId =
+      id || `checkbox-${label.replace(/\s+/g, "-").toLowerCase()}`;
+
+    return (
+      <Flex
+        aria-disabled={disabled || readonly}
+        data-disabled={disabled || readonly}
+        flexDirection="row"
+        gap="8"
+        ref={ref}
+        {...props}
+      >
+        <Box
+          style={{
+            alignSelf: "start",
+            display: "flex",
+            paddingTop: "4px",
+          }}
+        >
+          <Box asChild {...styles.indicatorRoot()}>
+            <RadixCheckbox.Root
+              checked={checked}
+              data-disabled={disabled || readonly}
+              defaultChecked={defaultChecked}
+              disabled={disabled || readonly}
+              id={checkboxId}
+            >
+              <RadixCheckbox.Indicator {...styles.indicator()}>
+                {checkboxIcon.indeterminate}
+                {checkboxIcon.checked}
+                {checkboxIcon.unchecked}
+              </RadixCheckbox.Indicator>
+            </RadixCheckbox.Root>
+          </Box>
+        </Box>
+
+        <RadixLabel.Root htmlFor={checkboxId} {...styles.rightSection()}>
```

Comment:
Need to wrap `RadixLabel.Root` with `Box` or `Flex` if we want to add style with sprinkles.